app/labor/rate_limit_checker.rb

```diff
@@ -13,18 +9,9 @@ class UploadRateLimitReached < StandardError; end
   class DailyFollowAccountLimitReached < StandardError; end
 
   def limit_by_action(action)
-    result = case action
-             when "comment_creation"
-               user.comments.where("created_at > ?", 30.seconds.ago).size > 9
-             when "published_article_creation"
-               user.articles.published.where("created_at > ?", 30.seconds.ago).size > 9
-             when "image_upload"
-               Rails.cache.read("#{user.id}_image_upload").to_i > 9
-             when "follow_account"
-               user_today_follow_count > self.class.daily_account_follow_limit
-             else
-               false
-             end
+    check_method = "check_#{action}_limit"
+    result = respond_to?(check_method, true) ? send(check_method) : false
```

Comment:
Note the `true` as second argument to `respond_to?` so it also considers private methods.

Reply:
TIL that `respond_to?` takes a second parameter!